client/src/GardenPlants/GardenPlantsList.js

```diff
@@ -0,0 +1,37 @@
+import {useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+import Api from '../Api';
+
+function GardenPlantsList(){
+    const [gardenplants, setGardenPlants] = useState([]);
+    useEffect(function(){
+      Api.gardenplants.index().then(response => setGardenPlants(response.data));  
+    }, []);
+
+    function onDelete(gardenplant){
+      if (window.confirm(`Are you sure you wish to delete ${gardenplant.name}?`)){
+        Api.gardenplants.delete(gardenplant.id).then(function(){
+          
+          const newGardenPlants = gardenplants.filter(s => s.id !== gardenplant.id);
+          setGardenPlants(newGardenPlants);
+        });
+      }
+    }
+    return(
+      //display
+        <main className="continer">
+            <h1>Garden Plants List</h1>
+            <Link className="btn btn-primary" to="/gardenplants/new">New</Link>
+            <ul>
+              {gardenplants.map(s =>(
+                  <li>
+                    <p><Link to={`/gardenplants/${s.id}/edit`}>{s.Garden.name} has {s.Plant.name}</Link></p>
```

Comment:
Note that here we display the Garden and Plant referenced by this GardenPlant...

Reply:
did you mean to put "has" on line 28?

Reply:
Yes- is that confusing for you? Have you tried running the code (now merged onto main), creating a few test Gardens and Plants, and then creating GardenPlant records? Try it out, and see if it makes sense what you see in the list- do you see how GardenPlant implements the MANY-TO-MANY relationship between Garden and Plant?